pytest_splunk_addon/splunk.py

```diff
@@ -437,6 +444,7 @@ def splunk(request, file_system_prerequisite):
         os.environ["SPLUNK_USER"] = request.config.getoption("splunk_user")
         os.environ["SPLUNK_PASSWORD"] = request.config.getoption("splunk_password")
         os.environ["SPLUNK_VERSION"] = request.config.getoption("splunk_version")
+        os.environ["SC4S_VERSION"] = request.config.getoption("sc4s_version")
```

Comment:
I am not sure we need to keep it here, maybe in `sc4s_docker`?

@mkolasinski-splunk @mbruzda-splunk thoughts?

Reply:
That is valid point, I would propose to go step further and move all splunk type related logic to `splunk_docker` and `splunk_external`, leacing `splunk()` like below:
`    @pytest.fixture(scope="session")
def splunk(request, file_system_prerequisite):
    """
    This fixture based on the passed option will provide a real fixture
    for external or docker Splunk

    Returns:
        dict: Details of the splunk instance including host, port, username & password.
    """
    splunk_type = request.config.getoption("splunk_type")
    LOGGER.info("Get the Splunk instance of splunk_type=%s", splunk_type)
    splunk_fixture = f"splunk_{splunk_type}"
    try:
        request.fixturenames.append(splunk_fixture)
        splunk_info = request.getfixturevalue(splunk_external)
    except Exception as e:
        raise Exception(f"Failed to get Splunk fixture ({splunk_fixture}): {e}")

    yield splunk_info
    `

Reply:
I like it.